benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -906,20 +908,27 @@ void GraphicsBenchmarkApp::ProcessKnobs()
     }
     pAllTexturesTo1x1->SetVisible(enableSpheres && (pKnobPs->GetValue() == SpherePS::SPHERE_PS_MEM_BOUND));
 
-    // Update sphere resources and mesh
-    if (setupSphereResourcesAndMeshes) {
-        SetupSphereResources();
-        SetupSphereMeshes();
-    }
-
-    // Rebuild pipelines
-    if (rebuildSpherePipeline) {
-        SetupSpheresPipelines();
-    }
-
-    // Update descriptors
-    if (updateSphereDescriptors) {
-        UpdateSphereDescriptors();
+    if (enableSpheres) {
+        // Update sphere resources and mesh
+        const bool spheresAreSetUp = mSpheresAreSetUp;
+        if (!spheresAreSetUp) {
+            // This creates all resources.
+            SetupSpheres();
+        }
+        else {
+            const bool requireMoreSpheres = sphereInstanceCountKnobChanged && (pSphereInstanceCount->GetValue() > mInitializedSpheres);
+            if (requireMoreSpheres) {
+                SetupSphereMeshes();
+            }
+            // Rebuild pipelines
+            if (alphaBlendKnobChanged || depthTestWriteKnobChanged) {
```

Comment:
Sphere pipeline seems to be dependent on some other knobs like PS, VS, vertex buffer: https://github.com/google/bigwheels/blob/4c9fbb05e77d384456594e6b9ea165b7d0bbcca2/benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp#L738-L747
Do we need to add those here, or perhaps just remove this since I think when we record a command buffer it will compile the necessary pipeline? 
@hysw, @angela28chen

Reply:
No, this part can be removed.

`SetupSpheresPipelines()` should be called exactly once, the appropriate pipeline will be built on demand from current knob values.

Reply:
Thanks @hysw  for confirming it. Updated.

Reply:
Removed.